### DISCUSSION TITLE
Supabase JS SDK: SQL Injection protection?

### DISCUSSION THREAD

**Discussion by MentalGear** [supabase/supabase#9777] [Questions] [answered]
I was just wondering if Supabase's (JS) SDK has preventive measures to avoid SQL Injection attacks, for example. 

`username = '; DROP TABLES users'`
```js
 const { data, error } = await supabase
    .from('users')
    .select( userName )
```

Or what best practices you recommend?

**Comment by steve-chavez** [comment, marked as answer]
There's no SQL input coming from the clients(hence no SQL injection possible). The libraries build an HTTP query string that is sent to PostgREST. See https://postgrest.org/en/stable/api.html.

**Reply by gambadavide96** [reply]
Hi @steve-chavez , sorry i would ask you a question.
Can you describe how javascript library builds HTTP string?
Thank you.

**Reply by GaryAustin1** [reply]
@GambaDavide96 
You can see generally the formats by using the dashboard API tab.  Select a table and operation then you can compare the javascript-js call to the BASH (http) call.
You could also use [PostgREST.org](https://postgrest.org/en/stable/api.html) to get url formats the REST API uses.
Also you could look in the supabase/postgrest-js repository at the actual code that builds the strings.
